juniper/src/executor/look_ahead.rs

```diff
@@ -367,6 +374,9 @@ pub trait LookAheadMethods<'sel, S> {
 
     /// Get an iterator over the children for the current selection
     fn children(&self) -> Vec<&Self>;
+
+    /// Get the parent type in case there is any for this selection
+    fn applies_for(&self) -> Applies;
```

Comment:
Should we really expose `Applies` struct here? Wouldn't it be better to return `Option<&str>` instead, leaving `Applies` an inner implementation detail?

Reply:
Yeah, makes more sense.